internal/registry/client_test.go

```diff
@@ -370,3 +372,122 @@ func TestLookupModuleNetworkError(t *testing.T) {
 		t.Fatal("unexpected error, got:", err)
 	}
 }
+
+func TestModuleLocation_readRegistryResponse(t *testing.T) {
+	server := test.Registry()
+	defer server.Close()
+
+	cases := map[string]struct {
```

Comment:
None of these test cases seem to test the X-Terraform-Get header. Is that tested elsewhere?

Reply:
Hey János, thanks for the feedback. [This case](https://github.com/opentofu/opentofu/blob/6967669d7939c5781809bcbdd8d15320bd2d5c1f/internal/registry/client_test.go#L388-L390) does cover the alpha registry protocol, i.e. the location read from the header.

Reply:
I'm more concerned with touching the existing logic despite there being no test cases for it. Typically, a stabilization test would make sure that the existing (legacy) functionality doesn't break.

Reply:
For context, the test cases cover the status quo behaviour as well as the one in line with the OpenTofu v1 protocol (location in the response body).